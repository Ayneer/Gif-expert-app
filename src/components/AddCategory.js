import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

    const [newCategory, setNewCategory] = useState('');

    const handleOnSubmit = e => {
        e.preventDefault();
        if(!newCategory.trim()){
            alert('Debe escribir una categoria');
            setNewCategory('');
            return;
        }
        setCategories(cats => [newCategory, ...cats]);
        setNewCategory('');
    }
    const handleCategoryInputChange = e => setNewCategory(e.target.value);

    return (
        <form onSubmit={handleOnSubmit}>
            <input type='text' id='input_category' value={newCategory} onChange={handleCategoryInputChange} placeholder='My new category name...' />
            <button type='submit' id='button_add_category'>Add category</button>
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
