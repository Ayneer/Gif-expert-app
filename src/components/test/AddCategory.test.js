import { shallow } from 'enzyme';
import AddCategory from '../AddCategory';


describe('Add Category Test Suit', () => {

    const setCategory = jest.fn(callback => callback([]));
    let wrapper = shallow(<AddCategory setCategories={setCategory} />);

    test('should show the component successfully', () => {
        const btnAddCategory = wrapper.find('button');

        const { type, id } = btnAddCategory.props();

        expect(type).toBe('submit');
        expect(id).toBe('button_add_category');
        expect(wrapper).toMatchSnapshot();
    });

    test('should send an alert when the category is empty', () => {
        jest.spyOn(window, 'alert').mockImplementation(() => { });

        const form = wrapper.find('form');

        form.simulate('submit', { preventDefault: jest.fn() });

        expect(window.alert).toBeCalledWith('Debe escribir una categoria');
        expect(setCategory).not.toBeCalled();
    });

    test('should send the category successfully', () => {
        jest.spyOn(window, 'alert').mockImplementation(() => { });

        const input = wrapper.find('input');

        input.simulate('change', { target: { value: 'Hola' } });

        expect(window.alert).not.toBeCalled();
    });

    test('should send the category successfully', () => {
        jest.spyOn(window, 'alert').mockImplementation(() => { });

        const form = wrapper.find('form');

        form.simulate('submit', { preventDefault: jest.fn() });

        expect(window.alert).not.toBeCalled();
        expect(setCategory).toBeCalled();
        expect(setCategory).toBeCalledWith(expect.any(Function));
    });

});
