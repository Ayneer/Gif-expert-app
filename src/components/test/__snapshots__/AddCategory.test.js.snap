// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Category Test Suit should show the component successfully 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    id="input_category"
    onChange={[Function]}
    placeholder="My new category name..."
    type="text"
    value=""
  />
  <button
    id="button_add_category"
    type="submit"
  >
    Add category
  </button>
</form>
`;
